core/aws.py

```diff
@@ -20,18 +23,55 @@
     _S3_CLIENT = client(
         "s3",
         region_name=Config.AWS_REGION,
-        endpoint_url=Config.AWS_ENDPOINT_OVERRIDE,
     )
 
 
-def upload_file(file: FileIO, bucket: str, object_name: str):
+def upload_file(file: IO, bucket: str, object_name: str):
     """Uploads a file to an S3 bucket.
 
     :param file: a readable file-like object
     :param bucket: bucket to upload to
     :param object_name: S3 object name
     """
+    file.seek(0)
     try:
         _S3_CLIENT.upload_fileobj(file, bucket, object_name)
     except (ClientError, EndpointConnectionError) as bucket_error:
         current_app.logger.error(bucket_error)
+
+
+def get_file(bucket: str, object_name: str) -> BytesIO | None:
+    """Retrieves a file from an S3 bucket.
+
+    :param bucket: bucket to retrieve from
+    :param object_name: S3 object name
+    :return: retrieved file as a BytesIO
+    """
+    file = BytesIO()
+    try:
+        _S3_CLIENT.download_fileobj(bucket, object_name, file)
+    except (ClientError, EndpointConnectionError) as bucket_error:
+        current_app.logger.error(bucket_error)
+        return None
+    file.seek(0)
+    return file
+
+
+def get_failed_file(failure_id: str) -> FileStorage | None:
+    """Gets a failed file from S3 via its failure_id.
+
+    The failure id is logged at the time of ingest failure and is used within the name of the file saved to S3.
+
+    :param failure_id: the corresponding failure id
+    :return: the failed file's name and contents as a BytesIO
+    """
+    response = _S3_CLIENT.list_objects_v2(Bucket=Config.AWS_S3_BUCKET_FAILED_FILES)
+    file_list = response["Contents"]
+    filename_match = next((file["Key"] for file in file_list if failure_id in file["Key"]), None)
+    if not filename_match:
+        return None
+    return FileStorage(
+        stream=get_file(Config.AWS_S3_BUCKET_FAILED_FILES, filename_match),
+        filename=filename_match,
+        content_type=EXCEL_MIMETYPE,
```

Comment:
is it always Excel file? Probs worth putting in the docstring or func name or something?